fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -2120,6 +2147,13 @@ Reference<IDatabase> MultiVersionApi::createDatabase(const char* clusterFilePath
 		lock.leave();
 
 		Reference<IDatabase> localDb = localClient->api->createDatabase(clusterFilePath);
+		if (clusterSharedStateMap.find(clusterFile) == clusterSharedStateMap.end() ||
```

Comment:
localDB is used only for retrieving database protocol version and listening for its updates. This logic should be rather executed. 

I am not completely sure, where is the best place to put this code. It could be updateDatabase method. We should consider the upgrade scenario, probably we should include protocol version into the key of the shared object map, so that we avoid reusing a shared state object across different versions of the databases.